DAPathfinder/GeneticAlgorithm/GeneticAlgorithm.swift

```diff
@@ -72,7 +72,7 @@ class GeneticAlgorithm
     }
     
     private func getParent(fromGeneration generation: [Route], withTotalDistance totalDistance: CGFloat) -> Route? {
-        let fitness = CGFloat(arc4random() / UINT32_MAX)
+        let fitness = CGFloat(Double(arc4random()) / Double(UINT32_MAX)) * (generation.count - 1)
```

Comment:
This won't compile, you're missing CGFloat(generation.count - 1) :)

Reply:
1) You are absolutely right :)
2) You are right, but `fitness>0` here is OK, since we compare it to `currentFitness`, which is a cumulative fitnesses, so "often" greater than 0. Nevertheless, as you suggest above, it would be a better idea to modify algorithm to compute `Route.fitness(with....:)`

Reply:
Yes, the `currentFitness` is a cumulative variable, but it was supposed to have a range identical to the range of the `Route.fitness(with...)`. If you don't modify the fitness function to have the same range as the cumulative variable the algorithm won't work properly.

For this current commit we can't really use `(generation.count - 1)` (maybe you forgot to commit some files?).

I'm willing to merge this into my master if you remove the `(generation.count - 1)`, I'm eager to hear your opinion on the optimised fitness function.

I really appreciate you taking an interest in this project, thanks a lot :)